### DISCUSSION TITLE
Have different level of log

### DISCUSSION THREAD

**Discussion by mart-e** [transifex/cli#160] [Ideas]
Hi,

When doing a `tx pull`, the new script is very verbose and does not allow to easily filter debug, info, warning and error messages.
```
# Getting info about resources

odoo-14.account - Getting info
odoo-14.account_debit_note - Getting info
odoo-14.account_check_printing - Getting info
odoo-14.account_edi - Getting info
odoo-14.account_edi_facturx - Getting info
odoo-14.account_debit_note - Done
...
odoo-14.fetchmail_gmail - Getting info
odoo-14.event_crm - Done
odoo-14.fetchmail_outlook - Getting info
odoo-14.event_sale - Done
odoo-14.fleet - Getting info
odoo-14.fetchmail_gmail - Resource odoo-14.fetchmail_gmail does not exist
odoo-14.gamification - Getting info
...
 Pulling files

odoo-14.account [af] - Pulling file
odoo-14.account [am] - Pulling file
odoo-14.account [ar] - Pulling file
odoo-14.account [az] - Pulling file
odoo-14.account [bg] - Pulling file
odoo-14.account [af] - User defined language 'af' does not belong to project, skipping
odoo-14.account [bn] - Pulling file
odoo-14.account [am] - User defined language 'am' does not belong to project, skipping
odoo-14.account [bs] - Pulling file
odoo-14.account [az] - User defined language 'az' does not belong to project, skipping
odoo-14.account [bs] - User defined language 'bs' does not belong to project, skipping
odoo-14.account [ca] - Pulling file
odoo-14.account [ckb] - Pulling file
odoo-14.account [bn] - Local file is newer than remote, skipping
odoo-14.account [bg] - Local file is newer than remote, skipping
...
```
while the log in the previous version of the CLI looked like:
```
tx INFO: Pulling translations for resource odoo-14.base (source: odoo/addons/base/i18n/base.pot)
tx INFO: Skipping 'bs' translation (file: odoo/addons/base/i18n/bs.po).
tx WARNING:  -> et: odoo/addons/base/i18n/et.po
tx INFO: Skipping 'sk' translation (file: odoo/addons/base/i18n/sk.po).
tx INFO: Skipping 'de' translation (file: odoo/addons/base/i18n/de.po).
tx WARNING:  -> sv: odoo/addons/base/i18n/sv.po
tx INFO: Skipping 'si' translation (file: odoo/addons/base/i18n/si.po).
tx WARNING:  -> ro: odoo/addons/base/i18n/ro.po
tx INFO: Skipping 'ca' translation (file: odoo/addons/base/i18n/ca.po).
tx INFO: Skipping 'es_BO' translation (file: odoo/addons/base/i18n/es_BO.po).
....
tx ERROR: Resource odoo-14.fetchmail_gmail doesn't exist on the server.
tx INFO: Pulling translations for resource odoo-14.fetchmail_outlook (source: addons/fetchmail_outlook/i18n/fetchmail_outlook.pot)
...
```
where it was easy to filter out info and warning messages.

Fetching the translations of our project, was 15k lines of INFO but only a few were actual WARNING and ERRORs.
With the new CLI, fetching the same project [migrated](https://github.com/odoo/odoo/pull/112130), it is now 31k lines where it's hard to spot the warning or errors.

**Comment by codegaze** [comment]
Hello @mart-e, could you please try adding the `--silent` flag and see if this works for you?
This is documented in the CLI commands help but we need to update the documentation to include it.

**Comment by mart-e** [comment]
Hi,

The `--silent` is more turning off everything but errors. I was hoping for a more `--log-level=warning` kind of setting but I guess it works.

**Comment by codegaze** [comment]
We don't have any current plans to enhance the logging, but since it sounds reasonable, I don't want to close it. 
I will move this to discussions to see if it gets more interest and not lose it as a future enhancement! :)

**Comment by Swyter** [comment]
@codegaze Hi, Thanos. I just upgraded to the Go version of the Transifex client after the old client stopped working very recently; it fails by printing hundreds of HTML and JS lines to the screen (I understand that the API endpoint had to be closed, but it's still kind of funky to see the older program do that instead of maybe showing a proper JSON error explaining where to go) and I agree with the original request.

My problem with the standard level of output is also that, seemingly due to how the buffering works, it's constantly clearing and recreating all lines, causing a dizzying level of flickering on Windows, and making the scrollbar not very useful.

If you want to easily test this, pull this repository and run this script on Windows, no dependencies required: https://github.com/tldmod/tldmod/blob/master/Languages/tx.cmd

---

The other thing is that I don't find half of the lines very interesting, the lines tend to show very redundant information.

![imagen](https://user-images.githubusercontent.com/690187/223398138-2e86adbf-9a82-48a1-ab98-5f97c38e3997.png)

**Reply by Swyter** [reply]
Here are some examples of the flickering, it's hard to explain how unergonomic this is in a plain `cmd.exe` window. Couple that with having a huge amount of lines and only being able to scroll back to the latest few, if any.

https://user-images.githubusercontent.com/690187/223403177-bc5aaa03-55eb-40af-9e8e-f54531d7121e.mp4

I think it would be great if only the latest/bottom-most line would be rewritten, leaving the rest of the lines working as normal. But I don't think I even need a progress bar.

**Reply by Swyter** [reply]
https://user-images.githubusercontent.com/690187/223407301-b2f0a8a3-3c90-455d-9a75-3bfa7d96f1c8.mp4

https://user-images.githubusercontent.com/690187/223407330-925f0bb1-3016-41fb-a8aa-2dfd3c324183.mp4

**Reply by Swyter** [reply]
You can see what happens when one tries to scroll back, as the program is essentially `cls`ing and recreating everything each frame the program stops and you get a black screen.
